client/src/components/core/ms-modal/MsPasswordInputModal.vue

```diff
@@ -0,0 +1,77 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+<template>
+  <ms-modal
+    :title="title"
+    :subtitle="subtitle"
+    :close-button="{ visible: true }"
+    :confirm-button="{
+      label: okButtonText || $t('PasswordInputModal.ok'),
+      disabled: password.length === 0,
+      onClick: confirm,
+    }"
+  >
+    <ms-password-input
+      :label="inputLabel || ''"
+      v-model="password"
+      @on-enter-keyup="confirm()"
+    />
+  </ms-modal>
+</template>
+
+<script lang="ts">
+import MsPasswordInputModal from '@/components/core/ms-modal/MsPasswordInputModal.vue';
+
+export interface GetPasswordOptions {
+  title: string;
+  subtitle?: string;
+  inputLabel?: string;
+  okButtonText?: string;
+}
+
+export async function getPasswordFromUser(options: GetPasswordOptions): Promise<string | null> {
+  const modal = await modalController.create({
+    component: MsPasswordInputModal,
+    canDismiss: true,
+    cssClass: 'text-input-modal',
+    componentProps: {
```

Comment:
I know I wrote the code, but you should be able to do `componentProps: options`, we don't need to redeclare every prop one by one.